src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -1524,7 +1524,7 @@ public int Compare(Symbol fst, Symbol snd)
                     node = node.Parent;
                 }
 
-                CSDiagnosticInfo info = NotFound(where, simpleName, arity, where.ToString(), diagnostics, aliasOpt, qualifierOpt, options);
+                CSDiagnosticInfo info = NotFound(where, simpleName, arity, (where as NameSyntax)?.ErrorDisplayName() ?? simpleName, diagnostics, aliasOpt, qualifierOpt, options);
                 return new ExtendedErrorTypeSymbol(qualifierOpt ?? Compilation.Assembly.GlobalNamespace, simpleName, arity, info);
```

Comment:
It looks like **lines** of code 1285, 1424, 1431 are also using syntax node as a diagnostics argument. Should they be changed too?